JokeGeneratorConsole/Program.cs

```diff
@@ -4,29 +4,355 @@ namespace JokeGenerator
+            string spacing = new String (' ', 22);
+            string borderline = new String ('═', 43);
+            Console.WriteLine(" ╔{0}═{0}╗", borderline);
+            Console.WriteLine(" ║{0}{1}{0}║", spacing, title);
+            Console.WriteLine(" ╚{0}═{0}╝", borderline);
+        }
+
+        private static void Error()
+        {
+            Console.SetCursorPosition(0, Console.CursorTop);
+            Console.Write(new String(' ', Console.BufferWidth));
+            Console.SetCursorPosition(0, Console.CursorTop - 1);
+        }
+
+        private static void StartCondition()
+        {
+            bool repeat = true;
+
+            while (repeat == true)
+            {
+                Console.Write("Press [s] to start or [esc] to exit anytime. ");
+                inputChar = read.Key();
+                if (Char.ToLower(inputChar) != 's')
+                {
+                    Error();
+                }
+                else
+                {
+                    repeat = false;
+                }
+            }
+            Console.WriteLine();
+        }
+
+        private static void ViewCategories()
+        {
+            bool repeat = true;
+            string[] separateCategory;
+
+            categoryList = load.Categories(ChuckNorrisCategoriesLink);
+            categoryList = categoryList.Replace("\"", "");
+            categoryList = categoryList.Replace(",", "] [");
+            separateCategory = categoryList.Split(" ");
+            while (repeat == true)
+            {
+                Console.Write("Would you like to view all possible categories?  (Y/N): ");
+                inputChar = read.Key();
+                if (Char.ToLower(inputChar) == 'n')
+                {
+                    repeat = false;
+                }
+                else if (Char.ToLower(inputChar) == 'y')
+                {
+                    Console.WriteLine("\n\nCategories:");
+                    if (separateCategory.Length != 1)
+                    {
+                        Console.WriteLine(separateCategory[0] + separateCategory[1] + separateCategory[2] + separateCategory[3] + separateCategory[4] + separateCategory[5] + separateCategory[6] + separateCategory[7]);
+                        Console.WriteLine(separateCategory[8] + separateCategory[9] + separateCategory[10] + separateCategory[11] + separateCategory[12] + separateCategory[13] + separateCategory[14] + separateCategory[15]);
+                    }
+                    else
+                    {
+                        Console.WriteLine("Database could not be loaded.");
+                    }
+                    repeat = false;
+                }
+                else
+                {
+                    Error();
+                }
+            }
+            Console.WriteLine();
+        }
+
+        private static void SelectCategory()
+        {
+            bool repeat = true;
+
+            while (repeat == true)
```

Comment:
Generally speaking looping should only be used when you need to loop because they are easy to make inefficient. 

In this method for example, you set up a while loop, ask for input, and then in every case you stop looping immediately. Either the user selects no, they select yes, or there is an error. The loop here is just clutter that doesn't actually affect code flow except to perhaps accidentally get stuck in an endless loop if the code is changed the wrong way.